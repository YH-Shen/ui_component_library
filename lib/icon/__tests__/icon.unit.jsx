import renderer from "react-test-renderer";
import React from "react";
import { mount } from "enzyme";
import Icon from "../icon";

describe("Icon tests", () => {
    it("successiful rendering", () => {
        const json = renderer
            .create(<Icon name="twitter" />)
            .toJSON();
        expect(json).toMatchSnapshot();
    });
    it("onClick", () => {
        const fn = jest.fn();
        const component = mount(<Icon name="twitter" onClick={fn} />);
        component.find("svg").simulate("click");
        expect(fn).toBeCalled();
        // expect(c.find("use").exists()).toEqual(false);
        // c.setProps({ loading: true });
        // expect(
        //     c.find("use").getDOMNode().getAttribute("xlink:href")
        // ).toEqual("#loading");
    });
});
