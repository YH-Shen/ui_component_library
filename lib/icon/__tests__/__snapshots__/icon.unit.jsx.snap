// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Icon tests successiful rendering 1`] = `
<svg
  className="yui-icon"
>
  <use
    xlinkHref="#twitter"
  />
</svg>
`;
